### ISSUE TITLE
Git hooks can no longer find node

### ISSUE THREAD

**Issue by CherryDT** [microsoft/vscode-remote-release#5850]
Issue Type: <b>Bug</b>

When Git is used through VSCode, Git hooks cannot find `node` anymore (`which node` returns nothing). This is a regression, it worked fine until the update that got installed today.

I'm using the `pre-commit` npm package, and `node` is installed via the `asdf` version manager.

To debug this, I added `set -x` to the start of [the `hook` script](https://github.com/observing/pre-commit/blob/master/hook).

**This is the output when committing via VSCode:**

```
> git -c user.useConfigOnly=true commit --quiet --allow-empty-message --file - --allow-empty
++ which node
++ which nodejs
++ which iojs
+ HAS_NODE=
+ [[ -z '' ]]
+ source_home_file .bash_profile
+ file=/home/david/.bash_profile
+ [[ -f /home/david/.bash_profile ]]
+ source_home_file .zshrc
+ file=/home/david/.zshrc
+ [[ -f /home/david/.zshrc ]]
+ source_home_file .bashrc
+ file=/home/david/.bashrc
+ [[ -f /home/david/.bashrc ]]
+ source /home/david/.bashrc
++ case $- in
++ return
++ which node
+ NODE=
++ which nodejs
+ NODEJS=
++ which iojs
+ IOJS=
+ LOCAL=/usr/local/bin/node
+ BINARY=
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -x /usr/local/bin/node ]]
+ [[ '' == *--dry-run* ]]
++ '' -e 'console.log(require.resolve('\''pre-commit'\''))'
./node_modules/pre-commit/hook: line 49: : command not found
+ '' ''
./node_modules/pre-commit/hook: line 49: : command not found
```

**This is the output when running the exact same command from the terminal:**
```
++ which node
+ HAS_NODE=/home/david/.asdf/shims/node
+ [[ -z /home/david/.asdf/shims/node ]]
++ which node
+ NODE=/home/david/.asdf/shims/node
++ which nodejs
+ NODEJS=
++ which iojs
+ IOJS=
+ LOCAL=/usr/local/bin/node
+ BINARY=
+ [[ -n /home/david/.asdf/shims/node ]]
+ BINARY=/home/david/.asdf/shims/node
+ [[ '' == *--dry-run* ]]
++ /home/david/.asdf/shims/node -e 'console.log(require.resolve('\''pre-commit'\''))'
+ /home/david/.asdf/shims/node /mnt/c/Users/david/proj/***/node_modules/pre-commit/index.js
pre-commit: 
pre-commit: No changes detected.
pre-commit: Skipping the pre-commit hook.
pre-commit: 
(reading log message from standard input)
```

Extension version: 0.58.5
VS Code version: Code 1.62.1 (f4af3cbf5a99787542e2a30fe1fd37cd644cc31f, 2021-11-05T10:57:55.946Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No
Remote OS version: Linux x64 4.4.0-19041-Microsoft
Remote OS version: Linux x64 4.4.0-19041-Microsoft
Remote OS version: Linux x64 4.4.0-19041-Microsoft

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 PRO 4750U with Radeon Graphics (16 x 1697)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.23GB (13.38GB free)|
|Process Argv|--crash-reporter-id 86814b6d-9362-4292-9bc7-43662930765a|
|Screen Reader|yes|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu-20.04|
|OS|Linux x64 4.4.0-19041-Microsoft|
|CPUs|AMD Ryzen 7 PRO 4750U with Radeon Graphics (16 x 1700)|
|Memory (System)|31.23GB (13.38GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu-20.04|
|OS|Linux x64 4.4.0-19041-Microsoft|
|CPUs|AMD Ryzen 7 PRO 4750U with Radeon Graphics (16 x 1700)|
|Memory (System)|31.23GB (13.41GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu-20.04|
|OS|Linux x64 4.4.0-19041-Microsoft|
|CPUs|AMD Ryzen 7 PRO 4750U with Radeon Graphics (16 x 1700)|
|Memory (System)|31.23GB (13.40GB free)|
|VM|0%|
</details>

**Comment by aeschli** [comment]
Maybe a dup of https://github.com/microsoft/vscode/issues/136809?

**Comment by CherryDT** [comment]
@aeschli Probably not, as this is WSL1.

**Comment by CherryDT** [comment]
I think it's because `.bashrc` is no longer executed here. I know this worked in the past though.

(I also noticed that in the terminal, instead `.profile` isn't executed, causing the AWS ElasticBeanstalk CLI to be no longer in `PATH`. I also think this worked in the past though, but I'm not sure if maybe EB changed the installation method.)

**Comment by CherryDT** [comment]
OK this is solved now. Two things happened in quick succession, not sure which one solved it.

First, I noticed that the shell setting was on `null` for some reason (never touched it since installing VSCode on this computer though!), I changed it to `bash`:

![image](https://user-images.githubusercontent.com/6010463/142470591-78cdfe1a-297d-4cba-ac48-9a1b020602ed.png)

Second, VSCode updated to 1.62.2.

Now, both the things I mentioned are working fine again.
